src/codegen/broutine.jl

```diff
@@ -487,16 +487,17 @@ function threaded_subspace_loop(f_kernel, ctx::BitContext, brt::BitRoutine)
     end
 
     for t in 1:n-1
-        tbody = subspace_step_expanded(idx->kernel(:($base + $idx)), lheads[t+1:end], index, ctx, brt) do x
-            subspace_locs = Expr(:tuple, :(1:$m...), [:(plain_locs[$(n-k+1)]) for k in 1:t]...)
+        # TODO: for n>=3, we must replace the symbol index(t+1) with the symbol "base" in lheads[t+2]
```

Comment:
Testcase for bugfix https://github.com/QuantumBFS/BQCESubroutine.jl/pull/35/commits/c65ef4b4aa648ec665c8e997f265d1084ef99e07#

```julia
# run Julia with 4 threads
using Test
using BenchmarkTools
using LinearAlgebra
using BQCESubroutine
using YaoLocations
using BQCESubroutine: threaded_basic_broutine!
using Polyester

        st = rand(ComplexF64, 1<<4);
        locs = BQCESubroutine.Locations([1, 3]);
        θ = 2π * rand();
        st0 = broutine!(copy(st), Val(:PSWAP), locs, θ);
        st1 = broutine!(copy(st), BQCESubroutine.BPSWAP(θ), locs);
        @test st0 ≈ st1
```